include/deal.II/multigrid/mg_coarse.h

```diff
@@ -329,6 +339,22 @@ MGCoarseGridApplySmoother<VectorType>::initialize (const MGSmootherBase<VectorTy
     (&coarse_smooth_,typeid(*this).name());
 }
 
+
+template<class VectorType>
+void
+MGCoarseGridApplySmoother<VectorType>::clear()
+{
+  coarse_smooth = nullptr;
+}
+
+
+template<class VectorType>
+MGCoarseGridApplySmoother<VectorType>::~MGCoarseGridApplySmoother()
+{
+  clear();
```

Comment:
on the second though, i guess i see what you mean now. Indeed, one can remove the destructor.